dcpy/connectors/socrata/metadata.py

```diff
@@ -0,0 +1,146 @@
+    samples = c.get("cachedContents", {}).get("top", [])
+    sample = "<FILL_ME_IN>"
+    dcp_type = None
+
+    if samples:
+        sample = samples[0].get("item")
+
+    if c["renderTypeName"] == "number":
+        if type(sample) == float:
+            dcp_type = "double"
+        elif type(sample) == int:
+            dcp_type = "integer"
+        else:
+            dcp_type = "<FILL_ME_IN>!"
+    else:
+        dcp_type = soc_types_to_dcp_types[c["renderTypeName"]]
+
+    dcp_col = {
+        "name": c["fieldName"],
+        "display_name": c["name"],
+        "description": c["description"],
+        "data_type": dcp_type,
+        "example": sample,
+    }
+    if c["renderTypeName"] == "multipolygon":
+        dcp_col["readme_data_type"] = "geometry"
+
+    ENTIRELY_ARBITRARY_MAX_STANDARDIZED_VALS = 10
+    if (
+        samples
+        and len(samples) < ENTIRELY_ARBITRARY_MAX_STANDARDIZED_VALS
+        and len(samples) == int(c["cachedContents"].get("cardinality"))  # type: ignore
+    ):
+        dcp_col["values"] = [
+            [s["item"], "<FILL_IN_THE_VALUE_DESCRIPTION!>"] for s in samples
+        ]  # type: ignore
+    return models.Column(**dcp_col)  # type: ignore
+
+
+def make_dcp_metadata(socrata_md):
+    columns = [make_dcp_col(c) for c in socrata_md["columns"]]
+    ds = models.Dataset(
+        name="primary_shapefile", type="shapefile", filename="shapefile.zip"
+    )
+
+    return models.Metadata(
+        name=socrata_md["resourceName"].lower(),
+        display_name=socrata_md["name"],
+        summary=socrata_md["description"],
+        description=socrata_md["description"],
+        tags=socrata_md["tags"],
+        each_row_is_a=socrata_md["metadata"]["rowLabel"],
+        columns=columns,
+        destinations=[
+            models.SocrataDestination(
+                id="socrata_prod",
+                four_four=socrata_md["id"],
+                datasets=["primary_shapefile"],
+                omit_columns=[],
+            )
+        ],
+        dataset_package={
+            "datasets": [
+                {
+                    "name": "primary_shapefile",
+                    "filename": "shapefile.zip",
+                    "type": "shapefile",
+                    "overrides": {},
+                }
+            ],
+            "attachments": [
+                a["filename"] for a in socrata_md["metadata"]["attachments"]
+            ],
+        },
+    )
+
+
+app = typer.Typer(add_completion=False)
+
+
+@app.command("placeholder")
+def _placeholder():
+    pass
+
+
+@app.command("export")
+def _export_socrata_metadata(
+    four_four: str,
+    output_path: Path = typer.Option(
+        None,
+        "--output-path",
+        "-o",
+        help="Path to export the metadata to",
+    ),
+):
+    logger.info(f"fetching metadata for {four_four}")
+    md = metadata.make_dcp_metadata(
+        pub.Dataset(four_four=four_four).fetch_metadata()
+    ).model_dump()
```

Comment:
Just talked about this in person, but might be able to specify `exclue_none` here? No worries if you'd prefer to be more explicit below anyways

Reply:
@fvankrieken yeah, let me investigate that. I'd also like to look into that for recipes, so that the `recipe.lock.yml` files aren't so bloated.

Reply:
@fvankrieken  implemented